docs/architecture.md

```diff
@@ -265,3 +265,41 @@ Associations based on the Security Policies. These additional tunnel ports are
 not used to send traffic to a remote Node - the tunnel traffic is still output
 to the default tunnel port (`tun0`) with OVS flow based tunneling. However, the
 traffic from a remote Node will be received from the Node's IPsec tunnel port.
+
+### Hybrid, NoEncap, NetworkPolicyOnly TrafficEncapMode
+Besides the default `Encap` mode, which always creates overlay tunnels among
+Nodes and encapsulates inter-Node Pod traffic, Antrea also supports other
+TrafficEncapModes including `Hybrid`, `NoEncap`, `NetworkPolicyOnly` modes. This
+section introduces these modes.
+
+* ***Hybrid*** When two Nodes are in two different subnets, Pod traffic between
+the two Nodes is encapsulated; when the two Nodes are in the same subnet, Pod
+traffic between them is not encapsulated, instead the traffic is routed from one
+Node to another. Antrea Agent adds routes on the Node to enable the routing
+within the same Node subnet. For every remote Node in the same subnet as the
+local Node, Agent adds a static route entry that uses the remote Node IP as the
+next hop of its Pod subnet.
+
+`Hybrid` mode requires the Node network to allow packets with Pod IPs are sent
+out from the Nodes' NICs.
+
+* ***NoEncap*** Pod traffic is never encapsulated. Antrea just assumes Node
+network can handle routing of Pod traffic across Nodes. Typically this is
+achieved by the Kubernetes Cloud Provider implementation which adds routes for
+Pod subnets to the Node network routers. Antrea Agent still creates static
+routes on the Node for remote Nodes in the same subnet, which is an optimization
```

Comment:
suggestion for clarity: "creates static routes on each Node"

Reply:
Done.